rust/static_or_dynamic_scope.rs

```diff
@@ -0,0 +1,12 @@
+
```

Comment:
Rust won't do this with functions because the scope of _x is unclear to the compiler, so it forces us to use closures which have much more explicit scoping rules.